### ISSUE TITLE
Andrii: Headless CMS, 3 weeks

### ISSUE THREAD

**Issue by andriivam** [lab-brussels-1/home#403]
# Week 3

## I Need Help With:
- I have some problem with the exercises as in Postman for me was more clear and easy to do it than in the script. Plus for some of them i'm not really sure what exactly i should use `sort`, `filter` etc.....
## What went well?
- Understanding what a content management system is, and why people use it
-  Perform a simple `http` call via postman to retrieve a piece of information
## What went less well?
- The connection inside of CMS is kind confusing but i think with some time it will be more clear
## Lessons Learned
- How to sort, filter, populate data in Postman and how to convert than in JS file 
## Sunday Prep Work
- Still doing exercises and studying suggested material

**Comment by danielhalasz** [comment]
> * I have some problem with the exercises as in Postman for me was more clear and easy to do it than in the script.

Postman is just there to test an API, but you will sadly have to write the code yourself so it can be used in the frontend.

**Comment by danielhalasz** [comment]
> * Plus for some of them i'm not really sure what exactly i should use `sort`, `filter` etc.....

well it depends if you want to sort (ascending or descending, based on price or alphabetical order, etc.) or if you want to filter the database and only included a subset of the items in the result

**Comment by andriivam** [comment]
# Week 2


## I Need Help With:

- I understand relationship in Strapi but at some point i have issues with that. Specially when need to create users. As well i don't know maybe i missed that during last class but i don't know how should i connected my Starpi collection with HTML(simply create file in `public` folder or it should be done in other way ). And same problem about `qs.js` i can use `qs.js` which we have in our study lenses or it should be created with Strapi


## What went well?

- I finished chat collection but i have some doubts but generally it was good.  

## What went less well?

- At some point i feel like i get lost. I have to much gabs in this module. Was trying to find nice tutorial to understand better but all what i find is already using  `React` so was really confused as i don't know yet React and it wasn't really helpful for me and i feel like waist this week as have so many time but couldn't do to much 


## Sunday Prep Work

- Trying to understand how i can create chat app

**Comment by danielhalasz** [comment]
> * understand relationship in Strapi but at some point i have issues with that. Specially when need to create users.

yes, it is a bit confusing. I think we have to differentiate between the users who can control the Strapi UI and users who use the JS application. I will look into this a bit more.

> * how should i connected my Starpi collection with HTML(simply create file in `public` folder or it should be done in other way )

well these are 2 different topics, not sure which one you ask about. one is to use the backend created by Strapi in the frontend code, by fetching data from it (if run locally, it is served on localhost:1337), the other question is hosting the frontend app within the Strapi installation, in the public folder. in this case, opening localhost:1337 (or the URL where the app is hosted) in the browser will automatically display the contents of the public folder

**Comment by danielhalasz** [comment]
> * Was trying to find nice tutorial to understand better but all what i find is already using  `React` so was really confused as i don't know yet React and it wasn't really helpful for me and i feel like waist this week as have so many time but couldn't do to much

yes, just follow until they get to React, that is where we are now..creating the database, the relationships and fetching data using Postman and simple JS queries.

**Comment by andriivam** [comment]
## Week 3

### What was not clear, where did you get stuck?
- For me this module was super interesting but at the same time little bit hard. Everything was clear but its kind my personal issue. i understand all parts separately but getting problems when i need to put everything together. I had small issue with server but thanks to @danielhalasz i fix its fast with `npm rebuild`.

### What was clear, what did you master?
- How to create collection in `Strapi`, install my project and run server. As well filtering, sorting and populating data from server.

### Where can you still use some help?
- I am still little confused with the relationship between `Strapi` collections. It's not so bad but sometimes i have doubt which one i should chose. But probably i can help out with `Postman`. For me its wasn't difficult and it's quite useful and helpful app.

### Where can you help others moving forwards?
- In this module be honest i asked more for help than help for others. One of this people @MMikhailova so would like to say huge thank you for her)
---

## The Course

### What can there be more of?

I wish this module could be more longer as its super exciting

### What can there be less of?

All was perfect

### What material were most helpful (from HYF or elsewhere)?

This time i just used suggested material and watched record video from our class

### Any suggestions for future classes?

Just want to say huge `THANK YOU` such amazing module)
